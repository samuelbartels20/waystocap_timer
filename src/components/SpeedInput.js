import React from 'react';
import { Button } from 'semantic-ui-react';
import PropTypes from 'prop-types';

const SPEEDS = [1, 1.5, 2];

function SpeedInput({ speed, onChange }) {
  const buttons = SPEEDS.map(s => (
    <Button key={s} active={s === speed} onClick={() => onChange(s)}>
      {s.toFixed(1)}X
    </Button>
  ));
  return <Button.Group>{buttons}</Button.Group>;
}

SpeedInput.propTypes = {
  speed: PropTypes.number,
  onChange: PropTypes.func,
};

export default SpeedInput;
