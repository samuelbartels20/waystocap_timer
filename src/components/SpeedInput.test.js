import React from 'react';
import SpeedInput from './SpeedInput';
import { shallow, mount } from 'enzyme';

it('renders without crashing', () => {
  shallow(<SpeedInput />);
});

it('activates button based on speed', () => {
  const input = shallow(<SpeedInput speed={2} />);
  expect(input).toMatchSnapshot();
});

it('changes correct speed', () => {
  const onChange = jest.fn();
  const input = mount(<SpeedInput onChange={onChange} />);

  input
    .find('Button')
    .filterWhere(b => b.text() === '2.0X')
    .simulate('click');
  expect(onChange.mock.calls.length).toBe(1);
  expect(onChange.mock.calls[0][0]).toBe(2);

  input.unmount();
});
