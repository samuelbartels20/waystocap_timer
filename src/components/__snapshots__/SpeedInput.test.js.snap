// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activates button based on speed 1`] = `ShallowWrapper {}`;
